ledger-core/v2/runner/executionevent/event.go

```diff
@@ -132,89 +143,113 @@ type saveAsChild struct {
 	prototype   reference.Global
 }
 
-func (e saveAsChild) Prototype() reference.Global {
+func (e callConstructor) Prototype() reference.Global {
 	return e.prototype
 }
 
-func (e saveAsChild) Arguments() []byte {
+func (e callConstructor) Arguments() []byte {
 	return e.arguments
 }
 
-func (e saveAsChild) Constructor() string {
+func (e callConstructor) Constructor() string {
 	return e.constructor
 }
 
-func (e saveAsChild) ParentObjectReference() reference.Global {
+func (e callConstructor) ParentObjectReference() reference.Global {
 	return e.parentObjectReference
 }
 
-func (e saveAsChild) ParentRequestReference() reference.Global {
+func (e callConstructor) ParentRequestReference() reference.Global {
 	return e.parentRequestReference
 }
 
-func (e saveAsChild) ConstructOutgoing(execution execution.Context) payload.VCallRequest {
-	panic(throw.NotImplemented())
+func (e callConstructor) ConstructVCallRequest(execution execution.Context) *payload.VCallRequest {
+	execution.Sequence++
+
+	return &payload.VCallRequest{
+		CallType:            payload.CTConstructor,
+		Caller:              e.parentObjectReference,
+		Callee:              reference.Global{},
+		CallSiteDeclaration: e.prototype,
+		CallSiteMethod:      e.constructor,
+		CallSequence:        execution.Sequence,
+		CallReason:          e.parentRequestReference,
+		KnownCalleeIncoming: reference.Global{},
```

Comment:
Who has to fill this field?

Reply:
![image](https://user-images.githubusercontent.com/495636/81387160-ec6f8780-911e-11ea-96c0-553e0b01c370.png)
rtfm

Reply:
in future, when we'll implement registration of requests - we'll return "registered request" response and send that field when we'll repeat it.